[WIP] Refactor Migration Guides (#386)
fluxcd/website: content/en/docs/migration/flux-guide.md

> ---

Reviewer: Can you add redirects for the renamed files into `static/_redirects` please?

design.
This may at times mean that you have to make adjustments to the way your
current cluster configuration is structured. If you are in this situation
and need help, refer to the [support page](https://fluxcd.io/support/).
{{% /alert %}}

## Prerequisites

> - Flux CLI

Author: need to refer to the Flux CLI installation doc

- Kubernetes cluster version **1.16** or newer
- ``kubectl`` version **1.18** or newer.

## GitOps migration

With this method, you follow the usual bootstrap procedure, which creates a new git repository, then move your manifests over.

Flux v2 offers an installation procedure that is declarative first
and disaster resilient.

Using the `flux bootstrap` command you can install Flux on a Kubernetes cluster and configure it to manage itself from a Git repository. The Git repository created during bootstrap can be used to define the state of your fleet of Kubernetes clusters.

For a detailed walk-through of the bootstrap procedure see the [installation
guide](../installation/_index.md).

{{% alert color="info" color="warning" title="'flux bootstrap' target" %}}
`flux bootstrap` must not be run against a Git branch or path
that is already being synchronized by Flux v1, as this will make
them fight over the resources. Bootstrap to a **new Git
repository, branch or path**, and continue with moving the
manifests.
{{% /alert %}}

> After installing Flux v2 on your cluster using bootstrap, you can delete the Flux v1 from your clusters and move the manifests from the Flux v1 repository to the bootstrap one.

Author: Logically this doesn't make sense

- What the method involves doing
```
With this method, you follow the standard bootstrap steps to install flux on your cluster. After following the bootstrap you then move your manifests over to the repository that was created during bootstrap.
```
- How to do the method
```
- run the bootstrap, create new repo
- warn for migration on a branch/path already sync
- remove flux v1
- migrate manifests from v1 repo to v2 repo
```
- Further reading
```
- Flux v1 Bootstrap doc
```

## In-place migration

{{% alert color="info" color="warning" %}}
For production use we recommend using the **bootstrap** procedure (see the [Gitops migration](#gitops-migration) section above).
{{% /alert %}}

With this migration method, you use the same repo for your flux V1 installation for flux v2.

This method installs flux v2 in a similar way to how you would install flux v1

### Flux read-only mode

> Assuming you've installed Flux v1 to sync a directory with plain YAMLs from a private Git repo:

Author: ```suggestion
This guide assumes you've installed Flux v1 to sync a directory with plain YAMLs from a private Git repo.

For example:
```

```sh
$ flux get kustomizations app
NAME  REVISION                                      SUSPENDED READY
app  main/5302d04c2ab8f0579500747efa0fe7abc72c8f9b False     True
```

### Flux with Slack notifications

Assuming you have configured Flux v1 to send notifications to Slack with
[FluxCloud](https://github.com/justinbarrick/fluxcloud).

Follow the [notifications guide]({{< relref "../guides/notifications.md" >}}) to setup slack notifications for Flux V2

> For more details on setting up notifications see

Author: ```suggestion
For more information on how to set up notifications see
```
